central/graphql/resolvers/components_v2.go

```diff
@@ -111,6 +111,9 @@ func (resolver *Resolver) componentsV2Query(ctx context.Context, query *v1.Query
 }
 
 func (resolver *Resolver) imageComponentDataStoreQuery(ctx context.Context, args IDQuery) (*imageComponentResolver, error) {
+	if features.PostgresDatastore.Enabled() {
```

Comment:
This is fine here, assuming you are planning a clean code split on components and that new code does not in any way invoke this func.

Reply:
That's the plan, if it ends up being relevant I'll tweak it in the PR for image component support